service/controller/azuremachinepool/handler/migration/azuremachinepool_test.go

```diff
@@ -0,0 +1,308 @@
+package migration
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	oldcapzexpv1alpha3 "github.com/giantswarm/apiextensions/v6/pkg/apis/capzexp/v1alpha3"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	capzexp "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1beta1"
+
+	"github.com/giantswarm/azure-operator/v5/service/unittest"
+)
+
+func TestEnsureCreatedAzureMachinePoolIsCorrect(t *testing.T) {
+	testCases := []struct {
+		name                    string
+		oldAzureMachinePoolFile string
+		newAzureMachinePoolFile string
+		expected                expected
+	}{
+		{
+			name:                    "case 0: New AzureMachinePool references not updated",
+			oldAzureMachinePoolFile: "old_azuremachinepool.yaml",
+			newAzureMachinePoolFile: "new_azuremachinepool_references_not_updated.yaml",
+			expected: expected{
+				newAzureMachinePoolModificationsCount: 0,
+				oldAzureMachinePoolDeleted:            false,
+			},
+		},
+		{
+			name:                    "case 1: New AzureMachinePool status not updated",
```

Comment:
New AzureMachinePool CR is has Meatdat and Spec fully set, so new handler will update Status.